code/createDeviceSettingsTable.m

```diff
@@ -0,0 +1,757 @@
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            Power_SettingsTable.action{entryNumber_Power} = 'Sense Config';
+            Power_SettingsTable.recNum(entryNumber_Power) = NaN;
+            Power_SettingsTable.time{entryNumber_Power} = HostUnixTime;
+            
+            % This populates information about the powerBands (coded
+            % values, not in Hz). The bipolar channels associated with the
+            % bands are defined above (from timeDomainChannels). Sense chan1: Bands
+            % 1-2, sense chan2: Bands 3-4, Sense chan3: Bands 5-6, Sense
+            % chan4: Bands 7-8
+            
+            % Settings will remain in powerChannels, TDsampleRate, and fftConfig until
+            % updated; TDsampleRate and fftConfig needed in later processing for
+            % determinig powerBands
+            powerChannels = currentSettings.SensingConfig.powerChannels;
+            Power_SettingsTable.powerBands{entryNumber_Power} = powerChannels;
+            
+            % Get sample rate for each TD channel; all TD channels have
+            % same Fs (or is listed as NaN)
+            for iChan = 1:4
+                TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+            end
+            TDsampleRates = unique(TDsampleRates);
+            currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
+            Power_SettingsTable.TDsampleRates{entryNumber_Power} = currentTDsampleRate;
+            
+            % Get fftConfig info if updated
+            if isfield(currentSettings.SensingConfig,'fftConfig')
+                fftConfig = currentSettings.SensingConfig.fftConfig;
+            end
+            Power_SettingsTable.fftConfig(entryNumber_Power) = fftConfig;
+            
+            entryNumber_Power = entryNumber_Power + 1;
+        end
+        
+        % If FFT updated in this record, create
+        % new row in FFT_SettingsTable and populate with metadata
+        if isfield(currentSettings.SensingConfig,'fftConfig')
+            HostUnixTime = currentSettings.RecordInfo.HostUnixTime;
+            FFT_SettingsTable.action{entryNumber_FFT} = 'Sense Config';
+            FFT_SettingsTable.recNum(entryNumber_FFT) = NaN;
+            FFT_SettingsTable.time{entryNumber_FFT} = HostUnixTime;
+            
+            % Settings will remain in fftConfig until updated
+            fftConfig = currentSettings.SensingConfig.fftConfig;
+            FFT_SettingsTable.fftConfig(entryNumber_FFT) = fftConfig;
+            entryNumber_FFT = entryNumber_FFT + 1;
+            
+            % Get sample rate for each TD channel; all TD channels have
+            % same Fs (or is listed as NaN)
+            for iChan = 1:4
+                TDsampleRates(iChan) = str2double(TDsettings(iChan).sampleRate(1:end-2));
+            end
+            TDsampleRates = unique(TDsampleRates);
+            currentTDsampleRate = TDsampleRates(~isnan(TDsampleRates));
```

Comment:
Same question as above. If two different sampling rates are possible and are used, the table will not tell which sampling rates belongs to which channel.

Technically you can look in the TD_SettingsTable, but then there would be no need for this field in this table.

Reply:
only one sample rate is possible